File: julia/gameoflife.jl
generations = 500
height = 50
width = 100

mutable struct Board
    cells::Array{Bool, 1}
    height::Int
    width::Int
end

function getcell(board::Board, row::Int, col::Int)
    if row >= 1 && row <= board.height && col >= 1 && col <= board.width
        return board.cells[((row-1) * board.width) + col]
    else
        return false
    end
end

function nextgeneration(board::Board)::Board
    newboard = Board(zeros(Bool, board.height*board.width), board.height, board.width)
    for i in 1:board.height
        for j in 1:board.width
            cell = getcell(board, i, j)
            adjacent = 0

            for n in -1:1
                for m in -1:1
                    if (n == 0 && m == 0)
                        continue
                    end
                    if getcell(board, i+n, j+m)
                        adjacent += 1
                    end
                end
            end

            if cell
                if adjacent < 2
                    cell = false
                end
                if adjacent > 3
                    cell = false
                end
            else
                if adjacent == 3
                    cell = true
                end
            end

            newboard.cells[((i-1)*board.width)+j] = cell
        end
    end

    return newboard
end

function draw(board::Board)
    for i in 1:board.height
        for j in 1:board.width
            if board.cells[((i-1)*board.width)+j]
                print("██")
            else
                print("  ")
            end
            flush(stdout)
        end
        println()
    end
end     

function rungame(height::Int, width::Int, generations::Int)
    board = Board(rand(Bool, height*width), height, width)

    for _ in 1:generations
        draw(board)
        print("\033[", board.height, "A")
        board = nextgeneration(board)
    end
end

rungame(height, width, generations)

File: php/gameoflife.php
#!/usr/bin/php -q
<?php

class Board {
    public $height;
    public $width;
    public $cells;

    public function __construct($height, $width, $cells) {
        $this->height = $height;
        $this->width = $width;

        if ($cells == null) {
            for ($i = 0; $i < $height * $width; $i++) {
                $this->cells[] = (bool) rand(0, 1);
            }

        } else {
            $this->cells = $cells;
        }
    }

    public function copy() {
        return new Board($this->height, $this->width, $this->cells);
    }

    public function draw() {
        for ($i = 0; $i < $this->height; $i++) {
            for ($j = 0; $j < $this->width; $j++) {
                if ($this->cells[($i * $this->width) + $j]) {
                    echo "██";
                } else {
                    echo "  ";
                }
            }
            echo "\n";
        }
    }

    public function get_cell($row, $col) {
        if (0 <= $row && $row < $this->height && 0 <= $col && $col < $this->width) {
            return $this->cells[($row * $this->width) + $col];
        } else {
            return false;
        }
    }
}

function next_generation($board) {
    $board_copy = $board->copy();

    for ($i = 0; $i < $board->height; $i++) {
        for ($j = 0; $j < $board->width; $j++) {
            $cell = $board_copy->get_cell($i, $j);
            $adjacent = 0;

            for ($n = -1; $n <= 1; $n++) {
                for ($m = -1; $m <= 1; $m++) {
                    if ($n == 0 && $m ==0) {
                        continue;
                    }
                    if ($board_copy->get_cell($i+$n, $j+$m)) {
                        $adjacent++;
                    }
                }
            }

            if ($cell) {
                if ($adjacent < 2) {
                    $cell = false;
                }
                if ($adjacent > 3) {
                    $cell = false;
                }
            } else {
                if ($adjacent == 3) {
                    $cell = true;
                }
            }

            $board->cells[($i * $board->width) + $j] = $cell;
        }
    }
}

function run_game($height, $width, $generations) {
    $board = new Board($height, $width, null);

    for ($i = 0; $i < $generations; $i++) {
        $board->draw();
        echo "\033[".$height."A";
        next_generation($board);
    }
}

$generations = 500;
$height = 50;
$width = 100;

run_game($height, $width, $generations);

?>